crates/pallets/starknet/src/lib.rs

```diff
@@ -111,35 +117,37 @@ pub mod pallet {
     use mp_starknet::transaction::types::{
         EventError, EventWrapper as StarknetEventType, StateDiffError, Transaction, TxType,
     };
+    use pallet_transaction_payment::OnChargeTransaction;
     use serde_json::from_str;
     use sp_core::{H256, U256};
     use sp_runtime::offchain::http;
-    use sp_runtime::traits::UniqueSaturatedInto;
+    use sp_runtime::traits::{DispatchInfoOf, PostDispatchInfoOf, UniqueSaturatedInto};
+    use sp_runtime::transaction_validity::InvalidTransaction::Payment;
+    use sp_runtime::transaction_validity::UnknownTransaction::Custom;
     use sp_runtime::DigestItem;
-    use starknet_api::api_core::{ClassHash, ContractAddress, Nonce};
+    use starknet_api::api_core::{ChainId, ClassHash, ContractAddress, Nonce};
+    use starknet_api::block::{BlockNumber, BlockTimestamp};
+    use starknet_api::deprecated_contract_class::EntryPointType;
     use starknet_api::hash::StarkFelt;
     use starknet_api::state::{StateDiff, StorageKey};
     use starknet_api::stdlib::collections::HashMap;
-    use starknet_api::transaction::EventContent;
+    use starknet_api::transaction::{Calldata, EventContent};
     use starknet_api::StarknetApiError;
     use types::{EthBlockNumber, OffchainWorkerError};
-    type ContractClassMapping = HashMap<ClassHash, ContractClass>;
-
-    use frame_support::pallet_prelude::*;
-    use frame_support::sp_runtime::offchain::storage::StorageValueRef;
-    use frame_support::traits::{OriginTrait, Time};
-    use frame_system::pallet_prelude::*;
 
     use super::*;
     use crate::message::{get_messages_events, LAST_FINALIZED_BLOCK_QUERY};
-    use crate::types::{ContractStorageKeyWrapper, EthLogs, NonceWrapper, StarkFeltWrapper};
+    use crate::types::{
+        ContractClassMapping, ContractStorageKeyWrapper, EthLogs, MaxTransactionsPendingBlock, NonceWrapper,
+        StarkFeltWrapper,
+    };
 
     #[pallet::pallet]
     pub struct Pallet<T>(_);
 
     /// Configure the pallet by specifying the parameters and types on which it depends.
     #[pallet::config]
-    pub trait Config: frame_system::Config {
+    pub trait Config: frame_system::Config + pallet_transaction_payment::Config {
```

Comment:
please add in the rust doc of the trait why this coupling with this pallet is needed